src/pymoca/backends/casadi/model.py

```diff
@@ -361,9 +363,19 @@ def simplify(self, options):
                             rows.append(cols)
                         symbols.append(old_var.symbol)
                         values.append(ca.vertcat(*[ca.horzcat(*[v.symbol for v in row]) for row in rows]))
-                        new_vars.extend(itertools.chain.from_iterable(rows))
+                        new_vars_extra = list(itertools.chain.from_iterable(rows))
+                        if l == 'inputs':
```

Comment:
I don't understand. We can also delay regular states or even algebraic states.

Reply:
Yes, but the test case was with inputs